src/main/scala/wdl4s/MemorySize.scala

```diff
@@ -0,0 +1,39 @@
+package wdl4s
+
+import wdl4s.parser.MemoryUnit
+
+import scala.language.postfixOps
+import scala.util.{Failure, Success, Try}
+import scalaz.Scalaz._
+
+object MemorySize {
+  val memoryPattern = """(\d+(?:\.\d+)?)\s*(\w+)""".r
+  def parse(unparsed: String): Try[MemorySize] = {
```

Comment:
as per what I was saying in person - I think the point at which one says "This is a full and complete unit of error checking, after this is will no longer be combined with other errors", that's the point to either convert to a Try or throw the exception.

Just one man's opinion